### ISSUE TITLE
Java Support?

### ISSUE THREAD

**Issue by Kylex87** [alexkrechik/VSCucumberAutoComplete#243]
Hi,

would like to know how I can this extension for step definitions that are defined in Java?
The issue https://github.com/alexkrechik/VSCucumberAutoComplete/issues/28 suggests, that all supported languages (including java) are possible by now but so far I only could see ruby,ts,js being examples.

I tried to just define it in the .vscode/settings.json with *.java but that didn't work. So I wonder, is it even possible right now and if not, is it planned?

**Comment by alexkrechik** [comment]
Many Java automations are using this extension. Could you please provide extension settings, step definitions and file pathes examples ?

**Comment by Kylex87** [comment]
Sure, setup looks like this:

> {
    "cucumberautocomplete.steps": [
        "src/main/java/de/abas/acceptanceTests/stepDef/\*.java",
    ],
    "cucumberautocomplete.syncfeatures": "src/test/resources/de/abas/acceptanceTests/features/\*.feature",
    "cucumberautocomplete.strictGherkinCompletion": true,
    "editor.quickSuggestions": {
        "comments": false,
        "strings": true,
        "other": true
    },
}
>
One file containing several step definitions is (example):
src/main/java/de/abas/acceptanceTests/stepDef/GeneralERPStepDefs.java
> @Given("^I set the fake date to \"([^\"]*)\"")
	public void i_set_the_fake_date_to(String date) {
   ...
}
>

And one feature file containing that step is
src/test/resources/de/abas/acceptanceTests/features/test.feature

> Feature: My Feature
Scenario: My Scenario
Given I set the fake date to "01.01.1990"
>

With that setup I dont have autocomplete, Go To etc. enabled. Syntax highlightning, formatting etc. works just fine.

**Comment by alexkrechik** [comment]
Hello, is issue still valid for you? I've started to look at it in release-3.14.0 scope, but was unable to reproduce. Such step looks ok for me, so looks like this is some configuration problem.

**Comment by alexkrechik** [comment]
Please reopen issue or write here with any additional details found.

**Comment by hfontanez1968** [comment]
I have the same problem. I cannot get this to work when using Java annotations to define the step definition pattern. I have a similar setup as Kylex87.

**Comment by andreasmarkussen** [comment]
I am using kotlin files without issues! Love it!

**Comment by alexkrechik** [comment]
Please write here if the issue still be present - unfortunately, I'm still unable to reproduce it.

**Comment by Sedatyf** [comment]
Hey guys! 

It's still a valid issue for me. I don't know what's wrong with my setup. 
I have this in my settings.json from VSCode (I accessed it with CTRL+, shortcut)
![image](https://user-images.githubusercontent.com/59012307/89016103-6e371180-d318-11ea-9d6b-ea7052ccf949.png)

And this is the result on my feature file
![image](https://user-images.githubusercontent.com/59012307/89017393-8871ef00-d31a-11ea-995e-1dc5d96db960.png)
(sorry for the french gibberish)

So it does not detect my steps in my java file. Even if I give a specific file in the path. And I don't have step completion when I start typing.

**Comment by tracey-ruark** [comment]
I've got this same issue.  Here is my .vscode/settings.json file:

> {
    "cucumberautocomplete.steps": [
        "src/main/java/com/inin/testing/cucumber/stepdefs/ui/**/*.java"
    ],
    "cucumberautocomplete.strictGherkinCompletion": true,
    "cucumberautocomplete.syncfeatures": "src/test/resources/com/inin/testing/features/**/*.feature",
    "cucumberautocomplete.gherkinDefinitionPart": "(Given|When|Then)\\(",
    "editor.quickSuggestions": {
        "comments": false,
        "strings": true,
        "other": true
    }
}

The feature file is:

> Feature: My Test Feature
  ...
  Scenario:
    Given I am an agent logged into the app

And the step definition is located in `src/main/java/com/inin/testing/cucumber/stepdefs/ui/LoginUserStepDefs.java`:

> @Given("^I am an agent logged into the app$")
    public void i_am_an_agent_logged_the_app() {...}

I'm not seeing any errors or warnings in the feature file about not being able to find the step definitions, however it cannot go to the definition.  I highlight the step in the feature file and hit F12 but nothing happens.

Any suggestions on things to try?

**Comment by alexkrechik** [comment]
I've tried and I have warned in the setup regarding only one escape symbol at `(Given|When|Then)\(`. All the everything works fine for me:
<img width="768" alt="Screenshot 2021-08-02 at 20 35 30" src="https://user-images.githubusercontent.com/5772406/127901578-9e80934d-1f65-495a-ab86-b4996c0703b7.png">
<img width="864" alt="Screenshot 2021-08-02 at 20 35 02" src="https://user-images.githubusercontent.com/5772406/127901588-927a0f10-e8e1-4a68-9e76-de67128b47f3.png">

**Comment by tracey-ruark** [comment]
Thanks for the quick reply.  It looks like the github comment editor put some extra escape characters into my comment.  Everything matches exactly the screenshot you've posted above but its still not finding any step definitions.

If you have any suggestions or things to try I'd be happy to give them a shot.

**Comment by tracey-ruark** [comment]
If anybody else runs across this same issue as I did I was able to find the cause.  The reason it didn't work for me is because I was using a multi-root workspace.  As soon as I opened my project's as a folder in an otherwise empty Vscode window the cucumber extension started working immediately.

[Link to vscode workspaces.](https://code.visualstudio.com/docs/editor/workspaces)

[Here is the existing ticket for multi-root workspace support.](https://github.com/alexkrechik/VSCucumberAutoComplete/issues/131)
